examples/applications/plot_out_of_core_classification.py

```diff
@@ -169,7 +170,9 @@ def progress(blocknum, bs, size):
         if _not_in_sphinx():
             sys.stdout.write("\r")
         print("untarring Reuters dataset...")
-        tarfile.open(archive_path, "r:gz").extractall(data_path)
+        # extractall can cause a tar slip
+        with closing(tarfile.open(archive_path, "r:gz")) as archive:
```

Comment:
Based on the [Tarfile docs](https://docs.python.org/3/library/tarfile.html#tarfile-objects)

> A [TarFile](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile) object can be used as a context manager in a [with](https://docs.python.org/3/reference/compound_stmts.html#with) statement. It will automatically be closed when the block is completed.

```suggestion
        with tarfile.open(archive_path, "r:gz") as archive:
```

This comment applies to the other diffs as well.